debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -307,6 +359,10 @@ protected void pollComplete(List<SourceRecord> batch) {
                     previousOutputMillis += millisSinceLastOutput;
                 }
             }
+            Document lastOffsetDocument = getLastOffsetDocument();
```

Comment:
What's the reasoning for converting it to `Document` instead of having the predicate working on the offset map?

Reply:
That's actually a good question. I was mostly focusing this around the fact that the only way to compare to offsets is have them as documents. However, there's no need to pass them around as documents until that time. I'll fix this.